net/can/can_recvmsg.c

```diff
@@ -253,31 +266,23 @@ static inline int can_readahead(struct can_recvfrom_s *pstate)
        * beginning of the I/O buffer chain.
        */
 
-      if (recvlen >= iob->io_pktlen)
-        {
-          FAR struct iob_s *tmp;
+      /* No trimming needed since one CAN/CANFD frame can perfectly
+       * fit in one iob
+       */
 
-          /* Remove the I/O buffer chain from the head of the read-ahead
-           * buffer queue.
-           */
+      FAR struct iob_s *tmp;
 
-          tmp = iob_remove_queue(&conn->readahead);
-          DEBUGASSERT(tmp == iob);
-          UNUSED(tmp);
+      /* Remove the I/O buffer chain from the head of the read-ahead
+       * buffer queue.
+       */
 
-          /* And free the I/O buffer chain */
+      tmp = iob_remove_queue(&conn->readahead);
+      DEBUGASSERT(tmp == iob);
+      UNUSED(tmp);
 
-          iob_free_chain(iob);
-        }
-      else
-        {
-          /* The bytes that we have received from the head of the I/O
-           * buffer chain (probably changing the head of the I/O
-           * buffer queue).
-           */
+      /* And free the I/O buffer chain */
 
-          iob_trimhead_queue(&conn->readahead, recvlen);
-        }
```

Comment:
The assumption is that one CAN frame always fits in one iob (default CONFIG_IOIB_BUFLEN). There is no need to trim head of iob chain since there is only one iob in chain. So `iob_remove_queue(&conn->readahead);` completely is only needed